test/runtests.jl

```diff
@@ -129,3 +128,28 @@ end
     end
     @test ret == 1.23
 end
+
+@testset "snapshot" begin
```

Comment:
If you can I would like more complicated tests that mirror how RAI expects this to be used.